### PR TITLE
fix cluster statistics

### PR DESCRIPTION
**Pull request by ObiWahn**
Allows always access to statistics:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4931/

https://github.com/arangodb/docs/pull/88 -- related PR

### CODE REVIEW

File: lib/ApplicationFeatures/V8SecurityFeature.cpp [hunk 1]
```diff
@@ -369,6 +371,16 @@ bool V8SecurityFeature::shouldExposeEnvironmentVariable(v8::Isolate* isolate,
                                 _environmentVariablesBlacklistRegex);
 }
 
+bool V8SecurityFeature::isAllowedToAccessSystemDatabase(v8::Isolate* isolate) const {
+  TRI_GET_GLOBALS();
+  TRI_ASSERT(v8g != nullptr);
+  auto const& sec = v8g->_securityContext;
+  auto vocbase = GetContextVocBasePointer(isolate);
+  bool isSystemDataBase = !vocbase || ( vocbase->name() == "_system" ); // If the vocbase is a nullptr we are in arangosh
```
Comment (dothebart) [inline lib/ApplicationFeatures/V8SecurityFeature.cpp]: can we have a fail early if-operation here?

File: lib/Rest/GeneralRequest.h [hunk 2]
```diff
@@ -84,6 +84,8 @@ class GeneralRequest {
         _requestContext(nullptr),
         _isRequestContextOwner(false),
         _authenticated(false),
+        _authActive(true), // the default is true as in some cases
+                           // not active implies authenticated
```
Comment (graetzer) [inline lib/Rest/GeneralRequest.h]: Why is it necessary to add a flag here? Authentication is either on or off for the entire server ?!

**Comment by ObiWahn** [comment]
Should this really be system / internal?
```
(rr) p *request                                                                                                                                                             
$21 = {_vptr.GeneralRequest = 0x55a1053567d0 <vtable for arangodb::HttpRequest+16>, _version = arangodb::rest::ProtocolVersion::HTTP_1_1,                                   
  _protocol = 0x55a10455df1d "http", _connectionInfo = {serverPort = 8529, clientPort = 39718, serverAddress = "::", clientAddress = "::1",                                 
    endpoint = "http+tcp://[::]:8529", endpointType = arangodb::Endpoint::DomainType::IPV6, encryptionType = arangodb::Endpoint::EncryptionType::NONE, sslContext = 0x0},   
  _clientTaskId = 156085022804973, _databaseName = "katzen", _user = "hund", _requestContext = 0x7fe8c003e220, _isRequestContextOwner = true, _authenticated = true,        
  _authActive = true, _authenticationMethod = arangodb::rest::AuthenticationMethod::JWT, _type = arangodb::rest::RequestType::GET,                                          
  _fullUrl = "/_admin/cluster/health?_=1560850331637", _requestPath = "/_admin/cluster/health", _prefix = "/", _suffixes = std::vector of length 3, capacity 4 = {"_admin", 
    "cluster", "health"}, _contentType = arangodb::rest::ContentType::JSON, _contentTypeResponse = arangodb::rest::ContentType::JSON,                                       
  _headers = std::unordered_map with 11 elements = {["accept-encoding"] = "gzip, deflate, br", ["referer"] = "http://localhost:8529/_db/katzen/_admin/aardvark/index.html", 
    ["content-type"] = "application/json", ["user-agent"] = "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.80 Safari/537.36",     
    ["connection"] = "keep-alive", ["accept-language"] = "en-US,en;q=0.9,de;q=0.8",
    ["authorization"] = "bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjEuNTYwODUwMzI5NTM4MDI0ZSs2LCJleHAiOjE1NjM0NDIzMjksImlzcyI6ImFyYW5nb2RiIiwicHJlZmVycmVkX3VzZXJuYW1lIjoiaHVuZCJ9.CPIx940aoiArfOv78R9_i18JN2ZXVzbRExK"..., ["host"] = "localhost:8529", ["accept"] = "*/*", ["x-arango-frontend"] = "true",                                  
    ["x-requested-with"] = "XMLHttpRequest"}, _values = std::unordered_map with 1 element = {["_"] = "1560850331637"}, _arrayValues = std::unordered_map with 0 elements}   
(rr) p _is
_isDir                     _isPrefix                  _isSystem                  _isolate                   
_isLocked                  _isPrivateuseValueSubtags  _isVariantSubtag           
(rr) p _isSystem 
$22 = true
```

File: lib/Rest/GeneralRequest.h (hunk 2 above)
Comment (ObiWahn) [reply lib/Rest/GeneralRequest.h]: Why should we include the server feature allover the place? Is there a reason why we do not simply set authenticated to true authentication is disabled?

**Comment by ObiWahn** [comment]
this looks ok
```
(rr) p *request
$99 = {_vptr.GeneralRequest = 0x55a1053567d0 <vtable for arangodb::HttpRequest+16>, _version = arangodb::rest::ProtocolVersion::HTTP_1_1, 
  _protocol = 0x55a10455df1d "http", _connectionInfo = {serverPort = 8529, clientPort = 39718, serverAddress = "::", clientAddress = "::1", 
    endpoint = "http+tcp://[::]:8529", endpointType = arangodb::Endpoint::DomainType::IPV6, encryptionType = arangodb::Endpoint::EncryptionType::NONE, sslContext = 0x0}, 
  _clientTaskId = 156085022804973, _databaseName = "katzen", _user = "hund", _requestContext = 0x7fe8c003e220, _isRequestContextOwner = true, _authenticated = true, 
  _authActive = true, _authenticationMethod = arangodb::rest::AuthenticationMethod::JWT, _type = arangodb::rest::RequestType::GET, 
  _fullUrl = "/_admin/aardvark/statistics/coordshort", _requestPath = "/_admin/aardvark/statistics/coordshort", _prefix = "/", 
  _suffixes = std::vector of length 4, capacity 4 = {"_admin", "aardvark", "statistics", "coordshort"}, _contentType = arangodb::rest::ContentType::JSON, 
  _contentTypeResponse = arangodb::rest::ContentType::JSON, _headers = std::unordered_map with 10 elements = {["accept-encoding"] = "gzip, deflate, br", 
    ["referer"] = "http://localhost:8529/_db/katzen/_admin/aardvark/index.html", 
    ["user-agent"] = "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.80 Safari/537.36", ["connection"] = "keep-alive", 
    ["accept-language"] = "en-US,en;q=0.9,de;q=0.8", 
    ["authorization"] = "bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjEuNTYwODUwMzI5NTM4MDI0ZSs2LCJleHAiOjE1NjM0NDIzMjksImlzcyI6ImFyYW5nb2RiIiwicHJlZmVycmVkX3VzZXJuYW1lIjoiaHVuZCJ9.CPIx940aoiArfOv78R9_i18JN2ZXVzbRExK"..., ["host"] = "localhost:8529", ["accept"] = "*/*", ["x-arango-frontend"] = "true", 
    ["x-requested-with"] = "XMLHttpRequest"}, _values = std::unordered_map with 0 elements, _arrayValues = std::unordered_map with 0 elements}
(rr) p _isSystem 
$100 = false
```

File: lib/Rest/GeneralRequest.h (hunk 2 above)
Comment (graetzer) [reply lib/Rest/GeneralRequest.h]: yes. Please do not add pointless flags to requests. Use "ExecContext::isAuthEnabled()" or simply  make it a convenience method checking `AuthenticationFeature`

**Comment by ObiWahn** [comment]
Just remove security checks for statistics.

Now the experiment has gone to: https://github.com/arangodb/arangodb/tree/experiment/permissions-example-statistics

I was beginning to pass information about the user authentication status into the JavaScriptSecuritContest. Eventually we need something like this. A more sophisticated system that brings together authentication/authorization and internal permissions for tasks that are a result of the used context (eg: internal task / query / REST request). At the moment it can happen, that for example a rest action triggers something internal, but we loose the information about the origin along the way. Then it becomes unclear if the task should be executed or not. The evolution of our internal permission handling is out of the scope of this PR as we want to release. That is why the changes on this branch have been reverted and have been replaced by the simplest solution possible.

File: lib/Rest/GeneralRequest.h (hunk 2 above)
Comment (ObiWahn) [reply lib/Rest/GeneralRequest.h]: That does not answer at all why requests can be unauthorized when the feature responsible for authorization is disabled. Passing the bool should not be necessary at all - I totally agree. But what is good about the monolithic design that constantly uses global state and requires each part of the code to know about the other.

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4931/

**Comment by ObiWahn** [comment]
https://github.com/arangodb/docs/pull/88

**Pull request merged**